R/MsBackendMetaboLights.R

```diff
@@ -0,0 +1,267 @@
+#' @title MsBackend representing MS data from MetaboLights
+#'
+#' @name MsBackendMetaboLights
+#'
+#' @aliases MsBackendMetaboLights-class
+#'
+#' @description
+#'
+#' The `MsBackendMetaboLights` retrieves and represents mass spectrometry (MS)
+#' data from metabolomics experiments stored in the
+#' [MetaboLights](https://www.ebi.ac.uk/metabolights/) repository. The backend
+#' directly extends the [MsBackendMzR] backend from the *Spectra* package and
+#' hence supports MS data in mzML, netCDF and mzXML format. Upon initialization
+#' with the `backendInitialize()` method, the `MsBackendMetaboLights` backend
+#' downloads and caches the MS data files of an experiment locally avoiding
+#' hence repeated download of the data.
+#'
+#' @section Initialization and loading of data:
+#'
+#' New instances of the class can be created with the `MsBackendMetaboLights()`
+#' function. Data is loaded and initialized using the `backendInitialize()`
+#' function with parameters `mtblsId`, `assayId` and `filePattern`. `mtblsId`
+#' must be the ID of a **single** (existing) MetaboLights data set. Parameter
+#' `assayId` allows to define specific *assays* of the MetaboLights data set
+#' from which the data files should be loaded. If provided, it should be the
+#' file names of the respective assays in MetaboLights (use e.g.
+#' `mtbls_list_files(<MetaboLights ID>, pattern = "^a_")` to list all available
```

Comment:
it's "a_" instead of "^a_" no  ?

Reply:
same below in the param definition

Reply:
with `"^"` the pattern needs to be at the beginning of the string. Using just `"a_"` it would list all files that have an `"a_"` somewhere in the name. `"^a_"` filters to files that really start with `a_<assay name>` (`"a_"` would also list `"i_some_file_a.txt"`).

Reply:
ah yes sorry i got confused ! make completely sense